proof.go

```diff
@@ -49,36 +49,28 @@ type Proof struct {
 	isMaxNamespaceIDIgnored bool
 }
 
-type jsonProof struct {
-	Start                   int      `json:"start"`
-	End                     int      `json:"end"`
-	Nodes                   [][]byte `json:"nodes"`
-	LeafHash                []byte   `json:"leaf_hash"`
-	IsMaxNamespaceIDIgnored bool     `json:"is_max_namespace_id_ignored"`
-}
-
 func (proof Proof) MarshalJSON() ([]byte, error) {
```

Comment:
[not blocking][silly question] do we need to maintain the `type Proof` defined in this file? It seems very similar to the Protobuf generated type in `pb/proof.pb.go` with a few exceptions:
- Start/End are int64 in the protobuf type instead of int
- IsMaxNamespaceIgnored is exported in the protobuf type

If we could remove the `type Proof` defined in this file, would that aid maintainability? I assume then we could marshal / unmarshal to JSON with https://pkg.go.dev/google.golang.org/protobuf/encoding/protojson

It doesn't have to be done in this PR b/c this is a fix and I'm suggesting a refactor.

My primary concern with maintaining two copies of the Proof type is that if we add a new field to the `proof.go` type, the compiler won't tell us to also add it to the protobuf Proof type.

Reply:
We wouldnt even need that protojson, it would already marshal/unmarshal to JSON just fine alone. 

The issue with this is that we cannot define methods onto pb.Proof since it is autogenerated and in a different package. Also: see my above comment. So I would keep type Proof